spi_mcux_lpspi: restore old driver and disable by default

AnElderlyFox opened the pull request:
**Description:**
This pull request restores the previous `MCUX LPSPI` driver to its original location and disables it by default, providing a stable alternative to the new driver.

**Problem:**
When using the `i.MX RT1062` with `Winbond W25N01GVZEIG NAND flash` on `zephyr:main`, built with `Zephyr SDK 0.17.4`, the `YAFFS2` filesystem mounts correctly with the `new LPSPI` driver, but the partition contents appear corrupted: files are displayed as directories, and data cannot be accessed.

Using the `old driver` resolves the issue, indicating that the `new driver` can behave unpredictably in some configurations. The problems likely manifest under high driver load, which makes them non-obvious and difficult to reproduce consistently. This unpredictability suggests that fixing or improving the `new driver` may require significant time.

Providing the `old driver` as a stable alternative ensures reliable operation during this period and allows users to compare the behavior of both drivers. This backup option is particularly useful for users who want the new features of the Zephyr RTOS but experience issues with the `new driver`.

**Summary of changes:**
- Restore old driver:
Restores the MCUX LPSPI driver from commit 42511c80bacee5f072fc64b0c9d6dc068005e31b (before relocation to NXP folder). This provides a proven, stable alternative while the new driver matures.

- Disable by default:
The old driver is disabled by default to prevent conflicts with SPI_NXP_LPSPI.

How to use old driver:
```
CONFIG_SPI_NXP_LPSPI=n
CONFIG_SPI_MCUX_LPSPI=y
```
No changes occur unless the new driver is explicitly disabled and the old driver is enabled via configuration.

AnElderlyFox committed:
spi_mcux_lpspi: revert old driver

Fixes #97270.
Revert the old MCUX LPSPI driver to its original location to allow users
to choose which implementation to use. The new driver remains under
development and has shown instability and functional issues in
some configurations. Providing the previous version ensures a stable
and reliable option for users who prefer or require it.
The restored driver was taken from commit
<42511c80bacee5f072fc64b0c9d6dc068005e31b>
(before it was moved to the new location).

Signed-off-by: Isaev Denis <anelderlyfox@yahoo.com>

decsny reviewed (changes requested):
we do not want to support and maintain two drivers. If you are having an issue with the current driver that is upstream, then make a bug report.

AnElderlyFox commented:
@decsny, it’s good that you don’t want to maintain the old driver — those who actually use it will take care of supporting it.

decsny commented:
@AnElderlyFox it's in everyone's interest if you just make a bug report normally for whatever issue you're having.

Commit:
spi_mcux_lpspi: disable driver by default

Disabled the MCUX LPSPI driver by default to avoid conflicts with
the SPI_NXP_LPSPI driver. Added a dependency to ensure that the
MCUX LPSPI driver is not enabled when the new driver is selected.

Signed-off-by: Isaev Denis <anelderlyfox@yahoo.com>
